internal/commands/server/k8s_e2e_test.go

```diff
@@ -964,6 +964,192 @@ func TestTCPMeshService(t *testing.T) {
 	testenv.Test(t, feature.Feature())
 }
 
+func TestHTTPRouteReferencePolicyLifecycle(t *testing.T) {
+	feature := features.New("http route reference policy").
+		Assess("http route controller watches reference policy", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			serviceOne, err := e2e.DeployHTTPMeshService(ctx, cfg)
+			require.NoError(t, err)
+
+			namespace := e2e.Namespaces(ctx)[0]
+			configName := envconf.RandomName("gcc", 16)
+			className := envconf.RandomName("gc", 16)
+			gatewayName := envconf.RandomName("gw", 16)
+			routeName := envconf.RandomName("route", 16)
+			routeNamespace := e2e.Namespaces(ctx)[1]
+			refPolicyName := envconf.RandomName("refpolicy", 16)
+
+			resources := cfg.Client().Resources(namespace)
+
+			gcc := &apigwv1alpha1.GatewayClassConfig{
+				ObjectMeta: meta.ObjectMeta{
+					Name: configName,
+				},
+				Spec: apigwv1alpha1.GatewayClassConfigSpec{
+					ImageSpec: apigwv1alpha1.ImageSpec{
+						ConsulAPIGateway: e2e.DockerImage(ctx),
+					},
+					UseHostPorts: true,
+					LogLevel:     "trace",
+					ConsulSpec: apigwv1alpha1.ConsulSpec{
+						Address: hostRoute,
+						Scheme:  "https",
+						PortSpec: apigwv1alpha1.PortSpec{
+							GRPC: e2e.ConsulGRPCPort(ctx),
+							HTTP: e2e.ConsulHTTPPort(ctx),
+						},
+						AuthSpec: apigwv1alpha1.AuthSpec{
+							Method:  "consul-api-gateway",
+							Account: "consul-api-gateway",
+						},
+					},
+				},
+			}
+			err = resources.Create(ctx, gcc)
+			require.NoError(t, err)
+
+			gc := &gateway.GatewayClass{
+				ObjectMeta: meta.ObjectMeta{
+					Name: className,
+				},
+				Spec: gateway.GatewayClassSpec{
+					ControllerName: k8s.ControllerName,
+					ParametersRef: &gateway.ParametersReference{
+						Group: apigwv1alpha1.Group,
+						Kind:  apigwv1alpha1.GatewayClassConfigKind,
+						Name:  configName,
+					},
+				},
+			}
+			err = resources.Create(ctx, gc)
+			require.NoError(t, err)
+
+			// FIXME: should this use a different port?
+			checkPort := e2e.HTTPFlattenedPort(ctx)
```

Comment:
Yeah, so pretty sure that due to the way we bind/port-forward things in the `kind` setup that you'll have issues spinning up this gateway/gateways in other tests (i.e. you should see errors in the deployment that gets created due to not being able to bind to the node port and the pod shouldn't get scheduled). Generally in the `e2e` tests, **_due to port binding, you basically need to specify a new port for each gateway that you want to provision and issue requests through_**.

This is solvable with something like `MetalLB` allowing for dynamic allocation of ip addresses and dropping our port-forwarding in `kind` (and our use of `useHostPorts`), but I'm hesitant to add that because:

1) It makes the setup all the more complicated, and
2) On Windows/Mac `MetalLB` doesn't work out of the box due to offloading Docker to an external VM that doesn't actually share the same network as the host (you'd have to install something like https://github.com/chipmk/docker-mac-net-connect to run the e2e tests locally)

Reply:
This feels like a thing we should scope out and take on at some point in the future if we can make the UX nice, but 👍 to just poking another hole for now.